src/core/api/track_manager.ts

```diff
@@ -505,6 +638,46 @@ export default class LanguageManager {
       }));
   }
 
+  /**
+   * Returns all available video tracks for a given Period, as an array of
+   * objects.
+   *
+   * @returns {Array.<Object>}
+   */
+  public getAvailableVideoTracks(period : Period) : ITMVideoTrackListItem[] {
+    const periodItem = getPeriodItem(this._periods, period);
+    const videoInfos = periodItem && periodItem.video;
+    if (videoInfos == null) {
+      return [];
+    }
+
+    const chosenVideoAdaptation = this._videoChoiceMemory.get(period);
+    const currentId = chosenVideoAdaptation && chosenVideoAdaptation.id;
+
+    return videoInfos.adaptations
+      .map((adaptation) => {
+
+        const representations = adaptation
+          .representations
+          .map((representation) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 571. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=285&issues=AWU4fywqOTG07BqbIIYI&open=AWU4fywqOTG07BqbIIYI)

[ISSUE_KEY]: AWU4fywqOTG07BqbIIYI